components/ILIAS/Init/classes/class.ilInitialisation.php

```diff
@@ -79,10 +79,19 @@ protected static function recursivelyRemoveUnsafeCharacters($var)
             $mod = [];
             foreach ($var as $k => $v) {
                 $k = self::recursivelyRemoveUnsafeCharacters($k);
-                $mod[$k] = self::recursivelyRemoveUnsafeCharacters($v);
+                if (!in_array($k, [ilCtrlInterface::PARAM_CMD_CLASS, ilCtrlInterface::PARAM_BASE_CLASS], true)) {
+                    $mod[$k] = self::recursivelyRemoveUnsafeCharacters($v);
+                } else {
+                    $mod[$k] = str_replace(
+                        array("\x00", "\n", "\r", "'", '"', "\x1a"),
```

Comment:
I suggest to put `array("\x00", "\n", "\r", "'", '"', "\x1a")` in a variable (e.g. `$unsafe_chars`) and use this variable in the last `return` statement, e.g. `return strip_tags(str_replace(array_merge($unsafe_chars, ["\\"]), '', $var));`